Common/src/geometry_structure.cpp

```diff
@@ -6804,6 +6780,152 @@ void CPhysicalGeometry::CompleteCommsAll(int nSends,
   
 }
 
+void CPhysicalGeometry::PrepareOffsets(unsigned long val_npoint_global) {
+  
+  /*--- Compute the number of points that will be on each processor.
+   This is a linear partitioning with the addition of a simple load
+   balancing for any remainder points. ---*/
+  
+  beg_node   = new unsigned long[size];
+  end_node   = new unsigned long[size];
```

Comment:
The arrays in question are only allocated once per geometry container / mesh read, but it is always good to have more checks. 

I also want to run this through valgrind again anyway, so I will give it a look.

Reply:
I tested it with one of the cases in the repo and it seems fine. It is just since the method is public, and anyone can call it from anywhere in the code, it would be good to make sure at least one of those fields is NULL before allocating.